docs/architecture/adr-003-subaccounts.md

```diff
@@ -0,0 +1,102 @@
+# ADR 3: Module SubAccounts
+
+## Changelog
+
+## Context
+
+Currently `ModuleAccount`s must be declared upon Supply Keeper initialization. In addition to this they don't allow for separation of fungible coins within an account.
+
+We want to support the ability to define and manage sub-module-accounts.
+
+## Decision
+
+We will use the type `ModuleMultiAccount` to manage sub-accounts of type `ModuleAccount`.
+A `ModuleMultiAccount` may have zero or more `ModuleAccount`s.
+Each sub-account will utilize the existing permissioned properties of `ModuleAccount`.
+`ModuleMultiAccount` and `ModuleAccount` have no pubkeys.
+There is no limit on the number of `ModuleAccount`s that a `ModuleMultiAccount` can have.
+A `ModuleMultiAccount` has no permissions since it cannot hold any coins.
+Its constructor returns a `ModuleMultiAccount` with no `ModuleAccount`s.
+A sub-account cannot be removed from a `MultiModuleAccount`.
+The account number assigned to sub-accounts will begin at 0 and be monotonically auto incrementing.
```

Comment:
An idea: each module subaccount could be indexed by a string (aka submodule account name). But thinking out loud maybe that's kinda irrelevant as we could just have named byte variables (aka `const FooSubAccount = 0x00`)

Reply:
this thought process led us to the proposed changes posted below